application/src/main/java/org/opentripplanner/service/paging/PagingService.java

```diff
@@ -183,6 +185,20 @@ private PageCursorFactory mapIntoPageCursorFactory(@Nullable PageType currentPag
       latestArrivalTime,
       searchWindowUsed
     );
+    // If the first search is an arrive by search, add the latest itinerary's start time to allow setting cursor information correctly
+    // for the page cursor of the next page.
+    if (
+      currentPageType == null &&
+      PageCursorFactory.resolvePageTypeForTheFirstSearch(itinerariesSortOrder) == PREVIOUS_PAGE
+    ) {
+      if (itineraries.size() > 0) {
+        factory = factory.withFirstSearchLatestItineraryDeparture(
+          itineraries.get(0).startTimeAsInstant()
+        );
+      } else {
+        factory = factory.withFirstSearchLatestItineraryDeparture(earliestDepartureTime);
+      }
+    }
```

Comment:
This block should be replaced with a private method - this is a very minor case, and now it totally dominates the logic in this method.

Reply:
I created a `handleFirstSearch` method